File: NaftaRozklad.Android/app/src/main/java/com/naftarozklad/repo/external/WebApi.kt
package com.naftarozklad.repo.external

import com.naftarozklad.repo.models.DayDTO
import com.naftarozklad.repo.models.Group
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

/**
 * Created by Bohdan.Shvets on 04.10.2017
 */
interface WebApi {

	@GET("api/groups.php")
	fun getGroups(): Call<List<Group>>

	@GET("api/schedules.php")
	fun getSchedule(@Query("group_id") groupId: Int, @Query("week") week: Int = 0, @Query("subgroup") subgroup: Int = 0): Call<List<DayDTO>>
}

File: NaftaRozklad.iOS/NaftaRozklad/DataAccess/WebApi/WebApi.swift
//
//  WebApi.swift
//  NaftaRozklad
//
//  Created by Yevhen Velizhenkov on 12/10/17.
//  Copyright © 2017 Yevhen Velizhenkov. All rights reserved.
//

import Alamofire
import PromiseKit

class WebApi {
  static let sharedInstance = WebApi()
  let baseUrl = "http://rozklad.nung.edu.ua/application"
  
  func getGroups() -> Promise<[Group]> {
    let url = "\(baseUrl)/api/groups.php"
    
    return Alamofire.request(url, method: .get).validate().responseData().then {
      guard let groups = try? JSONDecoder().decode([Group].self, from: $0) else {
        return Promise(value: [Group]())
      }
      return Promise(value: groups)
    }
  }
  
  func getSchedule(for group: Group, week: Week, subgroup: Subgroup) -> Promise<[DayDto]> {
    let url = "\(baseUrl)/api/schedules.php"
    let parameters: Parameters = ["group_id": group.id, "week": week, "subgroup": subgroup]
    
    return Alamofire.request(url, method: .get, parameters: parameters).validate().responseData().then {
      guard let days = try? JSONDecoder().decode([DayDto].self, from: $0) else {
        return Promise(value: [DayDto]())
      }
      return Promise(value: days)
    }
  }
  
  class func isConnectedToInternet() -> Bool {
    return NetworkReachabilityManager()!.isReachable
  }
}
